drivers/include/at.h

```diff
@@ -53,48 +54,71 @@ extern "C" {
 /**
  * @brief End of line character to send after the AT command.
  */
-#ifndef AT_SEND_EOL
-#define AT_SEND_EOL "\r"
+#if IS_ACTIVE(CONFIG_AT_SEND_EOL_WINDOWS)
+#define CONFIG_AT_SEND_EOL   "\r\n"
+#elif IS_ACTIVE(CONFIG_AT_SEND_EOL_UNIX)
+#define CONFIG_AT_SEND_EOL   "\n"
+#elif IS_ACTIVE(CONFIG_AT_SEND_EOL_MAC)
+#define CONFIG_AT_SEND_EOL   "\r"
+#endif
+
+#ifndef CONFIG_AT_SEND_EOL
+#define CONFIG_AT_SEND_EOL "\r"
+#endif
+
+/**
+ * @brief Enable this to disable check for echo after an AT
+ * command is sent.
+ */
+#ifdef DOXYGEN
+#define CONFIG_AT_SEND_SKIP_ECHO
 #endif
 
 /**
  * @brief Enable/disable the expected echo after an AT command is sent.
+ *
+ * @deprecated Use inverse @ref CONFIG_AT_SEND_SKIP_ECHO instead.
+ * Will be removed after 2021.01 release.
  */
 #ifndef AT_SEND_ECHO
+#if IS_ACTIVE(CONFIG_AT_SEND_SKIP_ECHO)
+#define AT_SEND_ECHO 0
+#else
 #define AT_SEND_ECHO 1
 #endif
+#endif
 
 /**
- * @brief 1st end of line character received (S3 aka CR character for a modem).
+ * @brief default OK reply of an AT device.
  */
-#ifndef AT_RECV_EOL_1
-#define AT_RECV_EOL_1   "\r"
+#ifndef CONFIG_AT_RECV_OK
+#define CONFIG_AT_RECV_OK "OK"
 #endif
 
 /**
- * @brief 1st end of line character received (S4 aka LF character for a modem).
+ * @brief default ERROR reply of an AT device.
  */
-#ifndef AT_RECV_EOL_2
-#define AT_RECV_EOL_2   "\n"
+#ifndef CONFIG_AT_RECV_ERROR
+#define CONFIG_AT_RECV_ERROR "ERROR"
 #endif
+/** @} */
 
 /**
- * @brief default OK reply of an AT device.
+1st end of line character received (S3 aka CR character for a modem).
  */
-#ifndef AT_RECV_OK
-#define AT_RECV_OK "OK"
+#ifndef AT_RECV_EOL_1
+#define AT_RECV_EOL_1   "\r"
 #endif
 
 /**
- * @brief default ERROR reply of an AT device.
+1st end of line character received (S4 aka LF character for a modem).
  */
-#ifndef AT_RECV_ERROR
-#define AT_RECV_ERROR "ERROR"
+#ifndef AT_RECV_EOL_2
+#define AT_RECV_EOL_2   "\n"
 #endif
-/** @} */
 
 /** Shortcut for getting send end of line length */
-#define AT_SEND_EOL_LEN  (sizeof(AT_SEND_EOL) - 1)
+#define AT_SEND_EOL_LEN  (sizeof(CONFIG_AT_SEND_EOL) - 1)
 
 #if defined(MODULE_AT_URC) || DOXYGEN
 #ifndef AT_BUF_SIZE
```

Comment:
It seems you are missing this option

Reply:
I don't understand. That's dependent on `CONFIG_AT_SEND_EOL` which is configurable.

Reply:
I mean `AT_BUF_SIZE`

Reply:
Should we consider module dependent configurations ?

Reply:
I don't see why we should not include this option when `MODULE_AT_URC` is active

Reply:
Okay. I shall add a Kconfig menu entry that will depend on `MODULE_AT_URC.`

Reply:
I don't think we need a menu in this case. Just something like:
```
config AT_BUF_SIZE
   int "Buffer size"
   depends on MODULE_AT_URC
```

Reply:
Yes. But since its buffer size I have forced exponential input.

Reply:
Updated the test results